File: core/src/main/java/com/cc/core/command/impl/OpenUrlCommandAction.java
package com.cc.core.command.impl;

import com.cc.core.actions.Action;
import com.cc.core.actions.ActionResult;
import com.cc.core.actions.Actions;
import com.cc.core.utils.Utils;
import com.cc.core.wechat.invoke.OpenUrlAction;

public class OpenUrlCommandAction implements Action {
    @Override
    public ActionResult execute(Object... args) {
        ActionResult result = Actions.execute(OpenUrlAction.class, args);
        if (!result.isSuccess()) {
            return result;
        }
        if (args.length > 1) {
            Utils.sleep((Long) args[1]);
        }
        return ActionResult.successResult();
    }

    @Override
    public String key() {
        return "openUrl";
    }
}

File: core/src/main/java/com/cc/core/command/impl/OpenUrlCommandAction.kt
package com.cc.core.command.impl

import com.cc.core.actions.Action
import com.cc.core.actions.ActionResult
import com.cc.core.actions.Actions
import com.cc.core.actions.accessibility.impl.BackAction
import com.cc.core.utils.Utils
import com.cc.core.wechat.invoke.OpenUrlAction

class OpenUrlCommandAction : Action {
    override fun execute(actiongId : String, vararg args: Any?): ActionResult? {
        val result = Actions.execute(OpenUrlAction::class.java, actiongId,  args)
        if (!result!!.isSuccess()) {
            return result
        }
        if (args.size > 1) {
            Utils.sleep((args[1] as Int).toLong())
            Actions.execute(BackAction::class.java, actiongId)
        }
        return ActionResult.successResult(actiongId)
    }

    override fun key(): String? {
        return "openUrl"
    }
}